### PR TITLE
Constable clothing and equipment for Security

### PR DESCRIPTION
**Pull request by Scrambledeggs00**
# Document the changes in your pull request

Adds a new uniform, two new hats and suits, a new neck item, and a new baton to Security themed around Constables. The clothing have the stats of basic security equipment, though the new neck item and baton are unique. The neck item is a whistle that can be blown through a UI action and the baton is a Detective's baton that does less damage in return for having a wrist strap that, when active, prevents you from dropping it. The clothing is available in the SecDrobe while the baton and whistle are only obtainable from the new Cargo crate "Constable Supply Crate" which is $5000.

I've tested all of this and, as far as I can tell, it all works.

# Spriting
Mob Sprites:
Constableclothing
Inhand Sprites:
ConstableInhand
Object Sprites:
ConstableObjects

# Wiki Documentation

Stats for wiki's combat page: The billy club does 10 brute damage, 15 stamina damage, and has a UI action slot that makes it undroppable while active.

Info for the supply crate: It is called "Constable Supply Crate" and is located in the Security section with a price of $5,000. It has two constable's uniforms, two constable's whistles, two billy clubes, one constable's vest, one constable's coat, one constable's hat, and one constable's helmet.

# Changelog

:cl:  Swims-In-Things for spriting advice and Medbay Lizard for helping with the whistle.
rscadd: Adds constable's uniform, coat, vest, helmet, hat, whistle, and billy club. 
soundadd: Adds a new sound for the whistle to blow people's eardrums out.
imageadd: Adds mob, object, and inhand sprites for all added items. 
/:cl:

### CODE REVIEW

**Comment by solwashere** [comment]
Shiggy innit did you pay for your tele license bruv

File: code/game/objects/items/melee/misc.dm [hunk 1]
```diff
@@ -508,6 +508,28 @@
...
+	if(active)
+		to_chat(usr, span_notice("You tighten the wrist strap."))
+		ADD_TRAIT(src, TRAIT_NODROP, WRIST_STRAP_TRAIT)
+	else
+		release_items()
+		to_chat(usr, span_notice("You untighten the wrist strap."))
+
+/obj/item/melee/classic_baton/secconbaton/proc/release_items()
+	REMOVE_TRAIT(src, TRAIT_NODROP, WRIST_STRAP_TRAIT)
+
```
Comment (tattax) [inline code/game/objects/items/melee/misc.dm]:
```suggestion
	var/tighten = FALSE
	actions_types = list(/datum/action/item_action/wrist_strap)

/obj/item/melee/classic_baton/secconbaton/ui_action_click(mob/user)
	tighten = !tighten
	if(tighten)
		to_chat(user, span_notice("You tighten the wrist strap."))
		ADD_TRAIT(src, TRAIT_NODROP, WRIST_STRAP_TRAIT)
	else
		REMOVE_TRAIT(src, TRAIT_NODROP, WRIST_STRAP_TRAIT)
		to_chat(user, span_notice("You untighten the wrist strap.")

```

File: code/modules/clothing/neck/_neck.dm [hunk 2]
```diff
@@ -464,3 +464,21 @@
...
+	var/recharge_rate = 50
+	actions_types = list(/datum/action/item_action/blow_whistle)
+
+/obj/item/clothing/neck/secconwhistle/ui_action_click(mob/user)
+	if(recharge_time > world.time)
+		to_chat(usr, span_warning("You need to catch your breath first!"))
+		return
+	playsound(loc, 'sound/misc/policewhistle.ogg', 30, 1, -1)
+	user.visible_message(span_warning("[user] blows their whistle!"))
+	recharge_time = world.time + recharge_rate
```
Comment (tattax) [inline code/modules/clothing/neck/_neck.dm]:
```suggestion
	COOLDOWN_DECLARE(recharge_time)
	var/recharge_rate = 5 SECONDS
	actions_types = list(/datum/action/item_action/blow_whistle)

/obj/item/clothing/neck/secconwhistle/ui_action_click(mob/user)
	if(!COOLDOWN_FINISHED(src, recharge_time)
		to_chat(user, span_warning("You need to catch your breath first!"))
		return
	playsound(get_turf(src), 'sound/misc/policewhistle.ogg', 30, TRUE, -1)
	user.visible_message(span_warning("[user] blows their whistle!"))
	COOLDOWN_START(src, recharge_time, recharge_rate)
```

**Comment by JamieD1** [comment]
Amazing. Can you make the hat less cross like?

**Comment by Scrambledeggs00** [comment]
> Amazing. Can you make the hat less cross like?

Object, Mob, or both sprites and what do you mean by "cross like"
Also are you talking about the hat on the far right or the one on the far left on the objects sprite image?

**Review by cuackles — APPROVED** [review]
so apart from the baton and whistle that have some banding and the seccon coat thats a bit to dark in my personal opinion everything else is fine, not gonna make a change request cus its not something i think its necesary to fix, would better if improved but it is not necesary

**Comment by ynot01** [comment]
@yogstation13/head-coders please rfeview https://github.com/yogstation13/Yogstation/blob/538ff5ee80a4c882e2244af0d4f637888dff471b/sound/misc/policewhistle.ogg

**Review by ToasterBiome — APPROVED** [review]

**Review by ynot01 — APPROVED** [review]
baton seems laughably weak but i dont see that as a reason to stop this from merging

File: TGS3.json
Comment (ynot01) [inline TGS3.json:1]: ?

Comment (Scrambledeggs00) [reply TGS3.json:1]: TBH I don't know what it is, but it wouldn't let me discard it and I couldn't switch to other branches on vscode until I committed it.

Comment (Scrambledeggs00) [reply TGS3.json:1]: Ok I think I deleted it in the merge fix

File: code/datums/actions/items/toggles.dm [hunk 3]
```diff
@@ -38,6 +38,12 @@
 /datum/action/item_action/toggle_welding_screen
 	name = "Toggle Welding Screen"
 
+/datum/action/item_action/blow_whistle
+	name = "Blow Your Whistle"
+
+/datum/action/item_action/wrist_strap
+	name = "Adjust Wrist Strap"
+
```
Comment (tattax) [inline code/datums/actions/items/toggles.dm]: if you want my opinion on it just move this to the files where the actions is actually run, i.e the whistle file and the baton one just above or below the item and their procs

File: code/game/objects/items/melee/misc.dm [hunk 4]
```diff
@@ -507,6 +507,25 @@
...
+	actions_types = list(/datum/action/item_action/wrist_strap)
+
+/obj/item/melee/classic_baton/secconbaton/ui_action_click(mob/user)
+	tighten = !tighten
+	if(tighten)
+		to_chat(user, span_notice("You tighten the wrist strap."))
+		ADD_TRAIT(src, TRAIT_NODROP, WRIST_STRAP_TRAIT)
+	else
+		REMOVE_TRAIT(src, TRAIT_NODROP, WRIST_STRAP_TRAIT)
+		to_chat(user, span_notice("You untighten the wrist strap."))
```
Comment (tattax) [inline code/game/objects/items/melee/misc.dm]: these to chats would work better as balloon alers btw

Comment (Scrambledeggs00) [reply code/game/objects/items/melee/misc.dm]: Thank you

**Pull request merged**